contrib/format-excel/src/main/java/org/apache/drill/exec/store/excel/ExcelBatchReader.java

```diff
@@ -348,6 +405,27 @@ private boolean nextLine(RowSetLoader rowWriter) {
     }
   }
 
+  private void populateMetadata() {
+    stringMetadata = new HashMap<>();
+    dateMetadata = new HashMap<>();
+
+    stringMetadata.put(IMPLICIT_STRING_COLUMN_NAMES.get(0), fileMetadata.getCategory());
```

Comment:
Please do not rely on the index of the specific column. You may define an enum with implicit excel field names and use its elements here and iterate through all its elements where required instead of using the list.

Reply:
Refactored with enum.